modules/lang-painless/src/main/java/org/elasticsearch/painless/node/SClass.java

```diff
@@ -159,6 +141,8 @@ public ScriptRoot analyze(PainlessLookup painlessLookup, CompilerSettings settin
             allEscape = statement.allEscape;
         }
 
+        scriptRoot.setUsedVariables(functionScope.getUsedVariables());
```

Comment:
Why is functionScope the container for these?  They used to be spread all over.

Reply:
Good question as this is in transition. It was easier to move it a single place for this PR since we only care about which function scope variables are used, but it turns out with removal of some of the custom code for the execute method in both trees it will be required throughout all scopes. This will be updated in future PRs, so I would prefer to not change it here.